src/migrate/index.js

```diff
@@ -119,10 +166,22 @@ export default class Migrator {
       .then((batchNo) => {
         return this._waterfallBatch(batchNo, migrations, direction)
       })
+      .then(() => this._unlockMigrations())
```

Comment:
Maybe just `.finally(() => this._unlockMigrations())` after the `.catch(...)` part. Then you won't need separate unlock in `.catch`block either.